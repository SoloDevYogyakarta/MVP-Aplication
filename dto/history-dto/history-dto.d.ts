export declare class CreateHistoryField {
    id?: number;
    name: string;
    title: string;
    desc: string;
    price: number;
    file_desc?: string;
    browse?: string;
}
