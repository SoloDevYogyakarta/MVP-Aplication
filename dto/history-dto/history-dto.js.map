{"version":3,"file":"history-dto.js","sourceRoot":"","sources":["../../../src/dto/history-dto/history-dto.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA,qDAAyE;AAEzE,MAAa,kBAAkB;CAyB9B;AAzBD,gDAyBC;AAtBC;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;8CACD;AAGZ;IADC,IAAA,0BAAQ,GAAE;;gDACE;AAGb;IADC,IAAA,0BAAQ,GAAE;;iDACG;AAGd;IADC,IAAA,0BAAQ,GAAE;;gDACE;AAIb;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;iDACC;AAId;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;qDACM;AAInB;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;kDACG"}