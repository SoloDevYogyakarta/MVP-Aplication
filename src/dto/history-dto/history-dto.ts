import { IsNumber, IsOptional, IsString } from 'class-validator';

export class CreateHistoryField {
  @IsString()
  name: string;

  @IsString()
  title: string;

  @IsString()
  desc: string;

  @IsNumber()
  @IsOptional()
  price: number;
}
