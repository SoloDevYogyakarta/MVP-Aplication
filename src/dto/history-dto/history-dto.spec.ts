import { faker } from '@faker-js/faker';
import { validate } from 'class-validator';
import { CreateHistoryField } from './history-dto';

describe('HistoryDto', () => {
  it('should to be defined', () => expect(CreateHistoryField).toBeDefined());

  it('render correctly', () => expect(CreateHistoryField).toMatchSnapshot());

  it('create', async () => {
    const data = new CreateHistoryField();
    data.name = faker.commerce.productName();
    data.title = faker.commerce.productMaterial();
    data.desc = faker.commerce.productDescription();
    data.price = faker.number.int({ min: 1000, max: 10000 });
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });
});
