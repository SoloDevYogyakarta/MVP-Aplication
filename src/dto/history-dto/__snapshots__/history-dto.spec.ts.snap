// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryDto render correctly 1`] = `[Function]`;
